### ISSUE TITLE
tf.summary.scalar TF 2.0 windows  empty user metric

### ISSUE THREAD

**Issue by sonfiree** [tensorflow/tensorboard#2753]
> Environment information (required)
> System: Windows 10, RAM 64 GB
> Environment: Python 3.7, TensorFlow-gpu
> GPU: GeForce RTX 1060 , CUDA 10.0, cudnn-10.0-windows10-x64-v7.4.2.24

```
logdir = join(
    "D:\\data\\tb_logs",
    datetime.datetime.now().strftime('%Y%m%d-%H%M%S'))

logdir_metrics = join(
    logdir,
    'metrics')

file_writer = tf.summary.create_file_writer(logdir_metrics)
file_writer.set_as_default()

t_Board = TensorBoard(log_dir= logdir, write_graph=True)
```

```
class Evaluate(Callback):
    def on_epoch_end(self, epoch, logs=None):
        tf.summary.scalar('test', data=2.0, step=epoch)
```

```
model.fit_generator(generator=tiger_train.next_batch() , 
                            steps_per_epoch=10,
                            epochs=10,
                            validation_data=tiger_val.next_batch(),
                            validation_steps=10,
                            callbacks=[t_Board, evaluator])
```

![image](https://user-images.githubusercontent.com/26105224/66542904-45594c00-eb4d-11e9-91b5-373e16a8eb6b.png)

**Comment by sonfiree** [comment]
` Method (on_train_batch_end) is slow compared to the batch update (0.413237). Check your callbacks.`

**Comment by davidsoergel** [comment]
Repro'd.  The issue is that your test data was buffered but never written to disk.  If you flush the writer, you'll see the data show up in TensorBoard.

    class Evaluate(tf.keras.callbacks.Callback):
        def on_epoch_end(self, epoch, logs=None):
            tf.summary.scalar('test', data=2.0, step=epoch)
            file_writer.flush()
